### DISCUSSION TITLE
Is a Tuya device with bluetooth ESP32?

### DISCUSSION THREAD

**Discussion by dashavoo** [arendst/Tasmota#17691] [Q&A] [answered]
I have been looking for an ESP32 based  UK smart plug. I have come across [this device on Amazon](https://www.amazon.co.uk/ALUSSO-Bluetooth-Wireless-Schedule-Monitoring/dp/B0BJ28MWQM/ref=mp_s_a_1_100) . I know Tuya started replacing ESP8266 board with another chip, but I don't know whether that affects devices with bluetooth. How good are the chances that this is an ESP32 device before I get one and try to open it?

Any tips for opening it also welcome. The devices with a similar form factor all seem to be ESP8266 and so just have instructions for using tuya-convert.

**Comment by barbudor** [comment, marked as answer]
Tuya has started to replace ESP chips everywhere
It is unlikely that any Tuya device start to use ESP32 has it is much above the price point of their other CPU provider

If you have the device in hand you don't need to open it to have confirmation:
- Start the device which may brodcast a Wifi network for you to connect to for configuration
- Find the MAC address of the device using a Wifi scanner such as [Net Analyser](https://play.google.com/store/apps/details?id=net.techet.netanalyzerlite.an) (Android, probably other exist for iPhone if that's your thing)
- Report the MAC address to a lookup site such as https://ouilookup.com/

That most likely give you the brand of the chip. Is not Espressif, just send it back
